src/transaction.rs

```diff
@@ -1112,5 +1111,337 @@ mod tests {
+            95d9470b9d7474d5582a29a0146d35ce27b534f29eb161d73e3baa18638d7077b\
+            e55251b3c321f0987ce59c53ac25eff5fb16df2009bd33a01cca1f16af5b78eb6\
+            3ae751c89f42b7b411b3d4787b7c7090f8f069f90b1b492926f4813e2a9e1c073\
+            9c073bea3ec2983ef713a880e33f5fa0b2f203f03fc6d9670bd44f6a27168fcdb\
+            9c538b2d8800b2f5d8f50b1b282d139a24f49d17942d1305042e2fc26116d6808\
+            16721cb2a6ecf0c7e68f85464766e832a0ac65601da66f00ca270b8db5d7f1d78\
+            9b518a5d2d27027be5d685b6df86c8e2ac7ec2f1b191c75bdfc0dc2b4b22c14bc\
+            3d828809adead46542c90cc3a0febd4391fd12cba8625de98d0b709b5d56efbd5\
+            80a0442687d0e677b8515fe7a47e2c002d1ee16c157c0d3f09b9ed14b948413d1\
+            b88056874395c24c9694781c108ce8aa6b0ac7889c977f030270985fa68be1d27\
+            74f4c7b9c06f1aab326301a56d424f5300c2c60c8220150f85192cde412823660\
+            66e87fe26d6ee5b7d284537d62dc698b5913f57c72b95c6ec43291274acb2c8f3\
+            109b82fad4b8a135b12163102f317b409a04f57e93ebe311d89a31cb17d338171\
+            45548c92a6339c988d1eee7d9d57475709c29f829ebb8ad5fe6113498b9ce6eb3\
+            91aa806d45fdcfac594b93b12c1e1635a6469248af357891c6bb24fee7cb38b65\
+            35af8462f90405e0f21ce4db4b20cb8c033c039c4e3561c3de87cc1687e0a6944\
+            ce9475179956392cbf2aa5c0ca9d487fbbb77d9a0856e1cc5b131b37f2e337a97\
+            e18b761c1fa5cb1a40845b5e2a12a7240ba050762e79aa754992218b092dfb22d\
+            a3fdaf9573589e767f12fdbe567593c581e9482209ee8c25258ee540b55e90c89\
+            8e7c834676db7b9b8e11b74049a4447d4267adddcaa85926a9ff56c128b29b895\
+            6423fc994355e90fb48caa17d4156b80bfe668cda0a0ec108a487f4a7193e25cb\
+            78fea170dc6956487744de0c263bd0c1847c5df09fad541b2be2d557896b566ae\
+            50186f922528705e5d8e7785f8ef9568f5edbb36e2d46ffc89b1b83439ff07ba4\
+            5c3d8f741d0000\
+        ");
+
+        assert_eq!(
+            tx.txid().to_string(),
+            "eda1d7c0f47fe209c3b5e98ec4bf48fc03f78ce8dcb9742683751fac42f7e4ed"
+        );
+        assert_eq!(tx.input.len(), 1);
+        assert_eq!(tx.output.len(), 3);
+        assert_eq!(tx.input[0].has_issuance, true);
+        assert_eq!(
+            tx.input[0].asset_issuance,
+            AssetIssuance {
+                asset_blinding_nonce: [0; 32],
+                asset_entropy: [0; 32],
+                amount: confidential::Value::Confidential(
+                    9,
+                    [
+                        0x81, 0x65, 0x4e, 0xb5, 0xcc, 0xd9, 0x92, 0x7b,
+                        0x8b, 0xea, 0x94, 0x99, 0x7d, 0xce, 0x4a, 0xe8,
+                        0x5b, 0x3d, 0x95, 0xa2, 0x07, 0x00, 0x38, 0x4f,
+                        0x0b, 0x8c, 0x1f, 0xe9, 0x95, 0x18, 0x06, 0x38
+                    ],
+                ),
+                inflation_keys: confidential::Value::Null,
+            }
+        );
+        assert_eq!(
+            tx.input[0].asset_entropy().unwrap().to_string(),
```

Comment:
travis complains that rustc doesn't know `asset_entropy()`

Reply:
Oh, whoops, I deleted that function (which is incomplete/blocked on bitcoin_hashes exposing midstates) but forgot to remove from the test.